tests/dao/project_dao_testing.rs

```diff
@@ -0,0 +1,43 @@
+extern crate proton_cli;
+
+use proton_cli::dao::ProjectDao;
+use proton_cli::error::Error;
+use proton_cli::project_types::Project;
+
+
+/// Implementation of ProjectDao for testing purposes. Uses given functions to return values.
+/// Functions are boxed so their sizes are known (pointers).
+/// The general naming convention used is trait_function_name_fn, for all trait functions.
+/// &str references are converted to Strings so we don't have to deal with lifetime headaches (bookdude13 tried on 12/25/16)
+pub struct ProjectDaoTesting {
+	pub new_project_fn: Box<Fn(String, u32) -> Result<Project, Error>>,
+	pub get_project_fn: Box<Fn(String) -> Result<Project, Error>>,
+	pub update_project_fn: Box<Fn(Project) -> Result<(), Error>>,
+}
+
+
+impl ProjectDaoTesting {
+	/// Creates a new ProjectDaoTesting struct with all functions set to return Error::TodoErr
+	pub fn new() -> ProjectDaoTesting {
+		ProjectDaoTesting {
+			new_project_fn: Box::new(|_, _| -> Result<Project, Error> { Err(Error::TodoErr) }),
+			get_project_fn: Box::new(|_| -> Result<Project, Error> { Err(Error::TodoErr) }),
+			update_project_fn: Box::new(|_| -> Result<(), Error> { Err(Error::TodoErr) })
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl ProjectDao for ProjectDaoTesting {
+	fn new_project(&self, name: &str, layoutid: u32) -> Result<Project, Error> {
```

Comment:
Not sure, but I think that's just GitHub's view of it. Doesn't look strange to me in Sublime...